core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -298,7 +310,17 @@ class BlockManagerMasterEndpoint(
     ).map(_.flatten.toSeq)
   }
 
-  private def register(id: BlockManagerId, maxMemSize: Long, slaveEndpoint: RpcEndpointRef) {
+  private def register(dummyId: BlockManagerId,
+    maxMemSize: Long,
```

Comment:
Can you also add a method doc saying this returns the same id with topology information attached?